sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -249,6 +249,8 @@ class JDBCOptions(
       .map(_.toBoolean)
       .getOrElse(SQLConf.get.timestampType == TimestampNTZType)
 
+  val hint = parameters.get(JDBC_HINT_STRING).map(_ + " ").getOrElse("")
```

Comment:
Please check the JDBC url starts with `jdbc:oracle` or `jdbc:mysql`.

Reply:
Ur, if you don't mind, can we keep the AS-IS code because there is no harm, @beliefer ? For me, there is no harm to have `hint` here, but please let me know your concerns.

Reply:
+1 for @beliefer 's comment to be robust.

Reply:
Sounds good, I'll make a update.